### ISSUE TITLE
This does not work with `eslint --report-unused-disable-directives`

### ISSUE THREAD

**Issue by ehmicky** [eslint/eslint-plugin-markdown#115]
`eslint --report-unused-disable-directives` is not reported when a `<-- eslint-disable -->` is defined in Markdown but is unused (i.e. could be removed).

`.eslintrc.yml`:

```yml
plugins: [markdown]
rules:
  no-inline-comments: 2
```

`README.md`:

````md
<!-- eslint-disable no-inline-comments -->

```js
var aa = true
```
````

`index.js`:

```js
/* eslint-disable no-inline-comments */
var aa = true
```

`package.json`:

```json
{
  "name": "eslint-comments-bug",
  "version": "1.0.0",
  "license": "ISC",
  "dependencies": {
    "eslint": "^5.16.0",
    "eslint-plugin-eslint-comments": "^3.1.1",
    "eslint-plugin-markdown": "^1.0.0"
  }
}
```

In the terminal:

```bash
$ eslint --report-unused-disable-directives README.md
$ eslint --report-unused-disable-directives index.js

/home/ehmicky/eslint-comments-bug/index.js
  1:1  error  Unused eslint-disable directive (no problems were reported from 'no-inline-comments')

✖ 1 problem (1 error, 0 warnings)
```

`--report-unused-disable-directives` works for the JavaScript file, but not for the Markdown (where nothing is reported).

**Comment by btmills** [comment]
I suspect the cause here is the same as in #75.

Before linting, the processor looks for HTML comments preceding a code block and inserts them as `/* ... */` comments at the beginning of the code. When it gets back a report from ESLint, it tries to map error locations in the linted code back to locations in the Markdown code block. Since the leading HTML comments aren't in the code block, if ESLint reports any errors on them, the plugin [ignores those messages](https://github.com/eslint/eslint-plugin-markdown/blob/a248c9abe7a3fe7c8c7d81d39440912d05ef8c49/lib/processor.js#L269).

A fix would replace the [bail-out check](https://github.com/eslint/eslint-plugin-markdown/blob/a248c9abe7a3fe7c8c7d81d39440912d05ef8c49/lib/processor.js#L269) for HTML comments and instead map error locations back to their original location above the code block. That would require tracking original location information for comments, which it doesn't do right now. A fix for either this or #75 should fix both. I'd welcome a contribution if you're interested in taking a stab at it, @ehmicky!

**Comment by ehmicky** [comment]
Thanks for your explanation!

I would gladly help but I am not familiar with writing ESLint plugins. It's on my to-do list but not yet unfortunately :(

**Comment by ehmicky** [comment]
I can confirm this is still a problem with the new `@eslint/markdown` and the new `--report-unused-disable-directives-severity` flag.

**Comment by andreahlert** [comment]
I've submitted a fix for this issue in #618

**Comment by xbinaryx** [comment]
Looks like #618 was closed, so I will work on this.
